### PR TITLE
update ldm noise scheduler api for inference

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes # .

### Description
update ldm noise scheduler api for inference

### Status
**Ready/Work in progress/Hold**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Comment by yiheng-wang-nv** [comment]
Thanks @Can-Zhao , I will add the unit tests on these 2 bundles within this PR to prevent similar issues.

**Comment by yiheng-wang-nv** [comment]
Hi @Nic-Ma , when preparing the unit test of ldm bundles, I just found a place of ldm bundles that is not matched up with:
https://github.com/Project-MONAI/MONAI/blob/e6ec945e4b87b90835cdad29ea64b1f27b8accda/monai/bundle/properties.py#L61

Some datasets may not have "data", such as all datasets in:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/apps/datasets.py
or customized datasets.

Is it possible to change it into `BundleProperty.REQUIRED: False` ? Or ldm bundles should extend `BundleWorkflow` directly and define a customized workflow (within the unit test) ?

**Comment by Nic-Ma** [comment]
Hi @yiheng-wang-nv ,

That's a good point, I would suggest to change the required `train_dataset_data` into `BundleProperty.REQUIRED: False` in the properties.
CC @SachidanandAlle and @tangy5 .

Thanks.

**Comment by yiheng-wang-nv** [comment]
> Hi @yiheng-wang-nv ,
> 
> That's a good point, I would suggest to change the required `train_dataset_data` into `BundleProperty.REQUIRED: False` in the properties. CC @SachidanandAlle and @tangy5 .
> 
> Thanks.

I see. Hi @SachidanandAlle @tangy5 , do you think we can change the "BundleProperty.REQUIRED" of `train_dataset_data`? Are there any concerns here?

File: models/brats_mri_axial_slices_generative_diffusion/configs/train_diffusion.json [hunk 1]
```diff
@@ -65,15 +65,15 @@
...
-    "inferer": {
-        "_target_": "generative.inferers.LatentDiffusionInferer",
-        "scheduler": "@noise_scheduler",
-        "scale_factor": "@scale_factor"
-    },
     "loss": {
         "_target_": "torch.nn.MSELoss"
     },
     "train": {
+        "inferer": {
```
Comment (yiheng-wang-nv) [inline models/brats_mri_axial_slices_generative_diffusion/configs/train_diffusion.json:72]: FYI @Can-Zhao , I changed to define "inferer" within "train" in order to align with:
https://github.com/Project-MONAI/MONAI/blob/858b86cf3af36e17efac8661533e3aa642dff5be/monai/bundle/properties.py#L66

**Review by yiheng-wang-nv — APPROVED** [review]
